server/src/main/java/com/alibaba/fescar/server/session/GlobalSession.java

```diff
@@ -467,4 +476,72 @@ public void decode(byte[] a) {
     public boolean hasBranch() {
         return branchSessions.size() > 0;
     }
+
+    public void lock() {
+        globalSessionSpinLock.lock();
+    }
+
+    public void unlock(){
+        globalSessionSpinLock.unlock();
+    }
+
+    public void lockAndExcute(LockRunnable excuteRunnable) throws TransactionException{
+        this.lock();
+        try {
+            excuteRunnable.run();
+        }finally {
+            this.unlock();
+        }
+    }
+
+    public <T> T lockAndExcute(LockCallable<T> lockCallable) throws TransactionException{
+        this.lock();
+        try {
+            return lockCallable.call();
+        }finally {
+            this.unlock();
+        }
+    }
+    /**
+     * globalsession is low race conditions, so use spinlock
+     */
+    private class GlobalSessionSpinLock{
+        /**
+         * true: Can lock, false : in lock.
+         */
+        private AtomicBoolean globalSessionSpinLock = new AtomicBoolean(true);
+
+        private static final int PARK_TIMES_BASE = 10;
+
+        public void lock() {
```

Comment:
Why not check the lock timeout?

Reply:
I don't think it's necessary，Since the optimistic lock is used, the locking time must be very short